jscomp/test/406_primitive_test.js

```diff
@@ -34,24 +34,20 @@ function f() {
   try {
     for (let i = 0; i <= 200; ++i) {
       if (i === 10) {
-        throw new Error(A, {
-          cause: {
-            RE_EXN_ID: A,
-            _1: 0
-          }
+        throw Caml_js_exceptions.internalFromExtension({
+          RE_EXN_ID: A,
+          _1: 0
         });
       }
       
     }
     return;
   } catch (raw_exn) {
-    let exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
+    let exn = Caml_js_exceptions.internalAnyToExn(raw_exn);
     if (exn.RE_EXN_ID === A) {
       return;
     }
-    throw new Error(exn.RE_EXN_ID, {
-      cause: exn
-    });
+    throw exn;
```

Comment:
I understand that this PR respects existing behavior. So this is just note for the future change

```suggestion
    throw raw_exn;
```

For example, suppose an exception raised in JS goes through a utility written in ReScript and then reaches JS again.

ReScript wraps all thrown values in RescriptError, regardless of whether it matches or not.
It can be problematic for FFIs because it can harm the intent of the external source.

Here's the ideal output I can think of:

```js
let A = Symbol.for("$$rescript$exn$Module.A")
try {
  // ...
} catch (raw) {
  if (RescriptError.isExn(raw, A)) {
    // ... just match, no implicit conversion
  }
  // rethrow value as-is
  throw raw;
}
```

Reply:
I agree

Reply:
I personally had the issue in rescript-schema. Had to solve with a hacky workaround https://github.com/DZakh/rescript-schema/blob/617ba9f38e1f06e0f4f98be029b78bcaa10898c7/src/S_Core.res#L413-L419